lib/client/data-manager.js

```diff
@@ -74,6 +74,36 @@ function DataManager(datasetId, $fh, dataSetNotificationStream, mediator) {
 }
 
 /**
+ * Sets sync status for a result object.
+ * @param {object} event - object passed in by sync event handler, should contain data used for logItem object.
+ */
+DataManager.prototype.addEvent = function addEvent(event) {
+  debug('addEvent', event);
+
+  var self = this;
+  var deferred = q.defer();
+  this.$fh.sync.getMetaData(this.datasetId, function(metadata) {
+    metadata = metadata || {};
+
+    metadata.syncEvents = metadata.syncEvents || {};
+
+    metadata.syncEvents[event.uid] = {
+      entityId: event.uid,
+      code: event.code,
+      action: event.message.action,
+      message: event.message.msg,
+      type: event.message.type,
+      ts: Date.now()
+    };
+
+    self.$fh.sync.setMetaData(self.datasetId, metadata, deferred.resolve, deferred.reject);
```

Comment:
I think we need something here that will clean up events. Otherwise, the data will always be stored.

I.e. when a remote_update is applied for a data item, there needs to be a check to see if there is any metadata stored for that item.